### ISSUE TITLE
Combining multiple VichImageType & VichFileType only submit the first field correctly

### ISSUE THREAD

**Issue by LeanderFS** [dustin10/VichUploaderBundle#952]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 1.8

#### Summary
I've created a form type with multiple VichImage & VichFile-type fields. I've created seperate mappings for all fields but when submitting the form only the first VichImage field posts a correct request. The second field is NULL, the third (FileType) doesn't upload but the temporary data seems to be posted.

I'm not sure if it matters, but I'm using Vich in combination with Gaufrette to upload the files to S3.

#### How to reproduce

Create a form using 2 or more VichFile or VichImage fields using unique mappings for all fields. Submit the form and check the POST data.

```yaml
knp_gaufrette:
    stream_wrapper: ~
    adapters:
        bookfairs_adapter:
            aws_s3:
                service_id: app.assets.s3
                bucket_name: '%s3_bucket_static%'
                detect_content_type: true
                options:
                    create: true
                    directory: test
    filesystems:
        bookfairs_fs:
            adapter:    bookfairs_adapter

vich_uploader:
    db_driver: orm
    storage: gaufrette
    mappings:
        bookfair_thumbnail:
            namer: AdminBundle\FileNamer\BookfairFileNamer
            directory_namer: AdminBundle\DirectoryNamer\BookfairDirectoryNamer
            inject_on_load: true
            uri_prefix: '%cloudfront_uri%test'
            upload_destination: bookfairs_fs
        bookfair_header:
            namer: AdminBundle\FileNamer\BookfairFileNamer
            directory_namer: AdminBundle\DirectoryNamer\BookfairDirectoryNamer
            inject_on_load: true
            uri_prefix: '%cloudfront_uri%test'
            upload_destination: bookfairs_fs
        bookfair_resource:
            namer: AdminBundle\FileNamer\BookfairFileNamer
            directory_namer: AdminBundle\DirectoryNamer\BookfairDirectoryNamer
            inject_on_load: true
            uri_prefix: '%cloudfront_uri%test'
            upload_destination: bookfairs_fs
```

```php
            ->add('thumbnailFile', VichImageType::class, [
                'allow_delete' => true,
                'required' => false,
                'constraints' => [
                    new Image([
                        'minWidth' => 300,
                        'minHeight' => 300,
                        'maxHeight' => 2500,
                        'maxWidth' => 2500,
                        'mimeTypes' => ['image/jpg', 'image/jpeg', 'image/png']
                    ])
                ],
                'label' => 'Choose a tile',
            ])
            ->add('headerFile', VichImageType::class, [
                'allow_delete' => true,
                'required' => false,
                'constraints' => [
                    new Image([
                        'mimeTypes' => ['image/jpg', 'image/jpeg', 'image/png']
                    ])
                ],
                'label' => 'Choose a header',
            ])
            ->add('resourceFile', VichFileType::class, [
                'allow_delete' => true,
                'required' => false,
                'label' => 'Choose a PDF',
            ])
```

```php
    /**
     * @var null|string
     */
    private $thumbnail;
    
    /**
     * @Vich\UploadableField(mapping="bookfair_thumbnail", fileNameProperty="thumbnail")
     * @var File
     */
    private $thumbnailFile;
    
    /**
     * @var null|string
     */
    private $header;
    
    /**
     * @Vich\UploadableField(mapping="bookfair_header", fileNameProperty="header")
     * @var File
     */
    private $headerFile;
    
    /**
     * @var null|string
     */
    private $resource;
    
    /**
     * @Vich\UploadableField(mapping="bookfair_resource", fileNameProperty="resource")
     * @var File
     */
    private $resourceFile;

    /**
     * @return string
     */
    public function getThumbnail(): ?string
    {
        return $this->thumbnail;
    }
    
    /**
     * @param string $thumbnail
     *
     * @return Bookfair
     */
    public function setThumbnail(?string $thumbnail): Bookfair
    {
        $this->thumbnail = $thumbnail;
        
        return $this;
    }
    
    /**
     * @return File
     */
    public function getThumbnailFile(): ?File
    {
        return $this->thumbnailFile;
    }
    
    /**
     * @param File $thumbnailFile
     *
     * @return Bookfair
     */
    public function setThumbnailFile(?File $thumbnailFile): Bookfair
    {
        $this->thumbnailFile = $thumbnailFile;
    
        if ($thumbnailFile) {
            $this->lastUpdate = new \DateTime('now');
        }
        
        return $this;
    }
    
    /**
     * @return string
     */
    public function getHeader(): ?string
    {
        return $this->header;
    }
    
    /**
     * @param string $header
     *
     * @return Bookfair
     */
    public function setHeader(?string $header): Bookfair
    {
        $this->header = $header;
        
        return $this;
    }
    
    /**
     * @return File
     */
    public function getHeaderFile(): ?File
    {
        return $this->headerFile;
    }
    
    /**
     * @param File $headerFile
     *
     * @return Bookfair
     */
    public function setHeaderFile(?File $headerFile): Bookfair
    {
        $this->thumbnailFile = $headerFile;
        
        if ($headerFile) {
            $this->lastUpdate = new \DateTime('now');
        }
        
        return $this;
    }
    
    /**
     * @return mixed
     */
    public function getResourceFile()
    {
        return $this->resourceFile;
    }
    
    /**
     * @param mixed $resourceFile
     *
     * @return Bookfair
     */
    public function setResourceFile(?File $resourceFile): Bookfair
    {
        $this->$resourceFile = $resourceFile;
    
        if ($resourceFile) {
            $this->lastUpdate = new \DateTime('now');
        }
    
        return $this;
    }
```

```
DefaultController.php on line 78:
Bookfair {#1700 ▼
  -id: "a52d831e-1af4-11e9-91cb-6003088c7080"
  -title: "test"
  -description: null
  -location: "test"
  -dateStart: DateTime @-27079747200 {#1693 ▶}
  -dateEnd: DateTime @-27079747200 {#1694 ▶}
  -preparationStart: DateTime @-61793107200 {#1695 ▶}
  -preparationEnd: DateTime @-27079747200 {#1696 ▶}
  -thumbnail: "AM18-tegel.jpeg"
  -thumbnailFile: UploadedFile {#15 ▶}
  -header: null
  -headerFile: null
  -resource: null
  -resourceFile: null
  -code: "AM18"
  -showOnSite: true
  -createdOn: DateTime @1547797366 {#1697 ▶}
  -lastUpdate: DateTime @1547798049 {#1779 ▶}
  -products: PersistentCollection {#1702 ▶}
  +"/private/var/tmp/php687fC5": UploadedFile {#16 ▶}
}
```

#### Expected behavior

All VichImage/VichFile fields get uploaded correctly.

**Comment by garak** [comment]
You need to define getters and setters for all your `File` properties and for all your mapped properties

**Comment by LeanderFS** [comment]
> You need to define getters and setters for all your `File` properties and for all your mapped properties

Yes I did that, I just didn't add them to the issue because I thought it would be redundant, but I'll add them right now.

**Comment by LeanderFS** [comment]
Unfortunatly that didn't work. I'll try to build a new form from the ground up to see if I made an error somewhere that I didn't notice.

Thanks for the fast response

**Comment by LeanderFS** [comment]
Wow, It seems I should stop coding early (5am) in the morning. After some coffee I ran trough the code again and noticed I made 2 errors. 

In my entity class, ```setHeaderFile``` points to the property ```thumbnailFile``` and the ```setResourceFile``` points to the property ```$this->$resourceFile``` (note the double variable).

I'm sorry for opening an issue for something that is obviously not an issue. I'll drink some coffee first the next time I run into an issue like this. Thanks for your time @garak, this issue can be closed.
